doc/classes/String.xml

```diff
@@ -232,30 +236,49 @@
 			</argument>
 			<description>
 				Converts a string containing a hexadecimal number into a decimal integer. If [code]with_prefix[/code] is [code]true[/code], the hexadecimal string should start with the [code]0x[/code] prefix, otherwise [code]0[/code] is returned.
-				[codeblock]
-				print("0xff".hex_to_int()) # Print "255"
-				print("ab".hex_to_int(false)) # Print "171"
-				[/codeblock]
+				[codeblocks]
+				[gdscript]
+				print("0xff".hex_to_int()) # Prints "255".
+				print("ab".hex_to_int(false)) # Prints "171".
+				[/gdscript]
+				[csharp]
+				GD.Print("0Xff".HexToInt()); // Prints "255".
+				// C# doesn't have `HexToInt(bool)`.
+				// GD.Print("ab".HexToInt(false)); // Prints "171".
```

Comment:
Is there a C# way around that limitation or should this method be added to C#, too?

Reply:
There is `int.Parse(instance.Substring(2), NumberStyles.HexNumber)` but we should just add it if desired.

This would be very easy to add, however I'm not sure why the core code is the way it is. The without-prefix version was not exposed in 3.2, and was only internally used in `String::is_valid_ip_address()`.

I think we should just have `hex_to_int` detect the prefix automatically and deal with it, so that we don't need a bool argument. `hex_to_int` already checks that the first two characters are `0x` and then does `s += 2;` if they exist. We can actually add support for this and simplify the code at the same time, by changing this:

```cpp
	if (p_with_prefix) {
		if (s[0] != '0' || s[1] != 'x') {
			return 0;
		}
		s += 2;
	}
```

To this:

```cpp
	if (s[0] == '0' && s[1] == 'x') {
		s += 2;
	}
```

Is there any reason not to do this? The same can be done for `bin_to_int`. @akien-mga